src/core/backbone_router/bbr_manager.cpp

```diff
@@ -98,61 +101,104 @@ void Manager::HandleMulticastListenerRegistration(const Coap::Message &aMessage,
     otError                    error     = OT_ERROR_NONE;
     bool                       isPrimary = Get<BackboneRouter::Local>().IsPrimary();
     ThreadStatusTlv::MlrStatus status    = ThreadStatusTlv::kMlrSuccess;
-    uint16_t                   addressesOffset, addressesLength;
-    Ip6::Address               address;
-    Ip6::Address               failedAddresses[kIPv6AddressesNumMax];
-    uint8_t                    failedAddressNum = 0;
-    TimeMilli                  expireTime;
     BackboneRouterConfig       config;
 
+    uint16_t     addressesOffset, addressesLength;
+    Ip6::Address address;
+    Ip6::Address failedAddresses[kIPv6AddressesNumMax];
+    uint8_t      failedAddressNum = 0;
+    TimeMilli    expireTime;
+
+    uint32_t timeout;
+    uint16_t commissionerSessionId;
+    bool     hasCommissionerSessionIdTlv = false;
+    bool     hasTimeoutTlv               = false;
+
     VerifyOrExit(aMessage.IsConfirmable() && aMessage.GetCode() == OT_COAP_CODE_POST, error = OT_ERROR_PARSE);
 
     VerifyOrExit(isPrimary, status = ThreadStatusTlv::kMlrBbrNotPrimary);
 
     // TODO: (MLR) send configured MLR response for Reference Device
-    // TODO: (MLR) handle Commissioner Session TLV
-    // TODO: (MLR) handle Timeout TLV
+
+    if (ThreadTlv::FindUint16Tlv(aMessage, ThreadTlv::kCommissionerSessionId, commissionerSessionId) == OT_ERROR_NONE)
+    {
+        const MeshCoP::CommissionerSessionIdTlv *commissionerSessionIdTlv =
+            static_cast<const MeshCoP::CommissionerSessionIdTlv *>(
+                Get<NetworkData::Leader>().GetCommissioningDataSubTlv(MeshCoP::Tlv::kCommissionerSessionId));
+
+        VerifyOrExit(commissionerSessionIdTlv != nullptr &&
+                         commissionerSessionIdTlv->GetCommissionerSessionId() == commissionerSessionId,
+                     error = OT_ERROR_PARSE);
+
+        hasCommissionerSessionIdTlv = true;
+    }
+
+    hasTimeoutTlv = hasCommissionerSessionIdTlv &&
```

Comment:
rename `hasTimeoutTlv` as `processTimeoutTlv`?
As here indicates the appearance of `ThreadTlv::kTimeout` is not equivalent to process it, right?